buf/registry/plugin/v1beta1/plugin.proto

```diff
@@ -0,0 +1,142 @@
+
+  // The id of the Plugin.
+  string id = 1 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.tuuid = true
+  ];
+  // The time the Plugin was created on the BSR.
+  google.protobuf.Timestamp create_time = 2 [(buf.validate.field).required = true];
+  // The last time the Plugin was updated on the BSR.
+  google.protobuf.Timestamp update_time = 3 [(buf.validate.field).required = true];
+  // The name of the Plugin.
+  //
+  // Unique within a given Organization.
+  string name = 4 [(buf.validate.field).string = {
+    min_len: 2
+    max_len: 100
+  }];
+  // The id of the Organization that owns the Plugin.
+  string owner_id = 5 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.tuuid = true
+  ];
+  // The Plugin's visibility, either public or private.
+  PluginVisibility visibility = 6 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The Plugin state, either active or deprecated.
+  PluginState state = 7 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The Plugin type.
+  PluginType type = 8 [
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum.defined_only = true
+  ];
+  // The configurable short description of the Plugin.
+  string description = 9 [(buf.validate.field).string.max_len = 350];
+  // The configurable source URL of the Plugin.
+  string source_url = 10 [
+    (buf.validate.field).string.uri = true,
+    (buf.validate.field).string.max_len = 255,
+    (buf.validate.field).ignore = IGNORE_IF_UNPOPULATED
+  ];
+  // Managed indicates whether a Plugin is managed by the BSR.
+  //
+  // Managed Plugins are automatically updated and should not be updated by users because they will
+  // be overwritten.
+  bool managed = 11;
+  // The collections the Plugin belongs to.
+  repeated Collection collections = 12;
+}
+
+// The visibility of a Plugin, currently either public or private.
+enum PluginVisibility {
+  PLUGIN_VISIBILITY_UNSPECIFIED = 0;
+  // PLUGIN_VISIBILITY_PUBLIC says that the plugin is publicly available.
+  PLUGIN_VISIBILITY_PUBLIC = 1;
+  // PLUGIN_VISIBILITY_PRIVATE says that the plugin is private.
+  PLUGIN_VISIBILITY_PRIVATE = 2;
+}
+
+// The state of a Plugin, currently either active or deprecated.
+enum PluginState {
+  PLUGIN_STATE_UNSPECIFIED = 0;
+  // PLUGIN_STATE_ACTIVE says that the Plugin is currently active.
+  PLUGIN_STATE_ACTIVE = 1;
+  // PLUGIN_STATE_DEPRECATED says that the Plugin has been deprecated and should no longer be used.
+  PLUGIN_STATE_DEPRECATED = 2;
+}
+
+// The type of the Plugin.
+enum PluginType {
+  PLUGIN_TYPE_UNSPECIFIED = 0;
+  // PLUGIN_TYPE_PROTOC_CODE_GENERATION says that the Plugin is a protoc code generation plugin. It
+  // is intentionally a separate type because it does not use the pluginrpc protocol.
+  PLUGIN_TYPE_PROTOC_CODE_GENERATION = 1;
```

Comment:
I believe we explicitly said in our discussions we did not want the API representing old protoc plugins, am I misremembering? Why or why not?

Reply:
We had a conversation about this - can you remind me on what we concluded and why?

Reply:
We initially said we wanted to unify all plugins into a common "plugin" entity (starting with protoc and check) since they share the same namespace and would reduce FE complexity. But then quickly realized we'd have to think through all the generated sdk config definitions and it was unclear how that would evolve when adding future generate plugins.

Furthermore, there were outstanding questions on whether the old protoc plugins would be updated to commits/labels in the UI (they are currently version/revision-based). This was after we decided that _new_ plugins will adopt a commit/label model instead of version/revision.

I do not remember us making a final decision on the API though, @emcfarlane @smallsamantha if you remember differently.

I propose we focus on check plugins in this new API to unblock new feature work like:

1. uploading check plugins and introducing `buf plugin push`
2. downloading check plugins and referencing them in `buf.yaml` files, unlocking shareable custom lint rules
3. introducing BSR features that use check plugins and custom lint rules
4. introducing the `buf registry plugin` set of commands, to have parity with `buf registry module`